### PR TITLE
fix(ecsutil): guard parseRegion against invalid TaskARN

### PR DESCRIPTION
**Pull request by bhavya76**
# Description of the issue
parseRegion in translator/util/ecsutil panics with an index out of range when the ECS metadata endpoint returns a TaskARN with fewer than 4 colon-separated segments.

# Description of changes
* parseRegion: return after logging when the split TaskARN has fewer than 4 colon-separated segments
* Add ecsutil_test.go covering valid and invalid TaskARN inputs for parseRegion

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
* Added unit tests in ecsutil_test.go
* `go test ./translator/util/ecsutil/` passes locally

# Requirements
_Before commiting your code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

-------
### Integration Tests
To run integration tests against this PR, add the `ready for testing` label.

### CODE REVIEW

File: translator/util/ecsutil/ecsutil_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+func TestParseRegion_ValidARN_LongFormat(t *testing.T) {
+	// Long-form ARN includes the cluster name segment before the task id.
+	e := &ecsUtil{}
+	em := &ecsMetadataResponse{TaskARN: "arn:aws:ecs:eu-west-2:123456789012:task/my-cluster/abc123"}
+	e.parseRegion(em)
+	assert.Equal(t, "eu-west-2", e.Region)
+}
+
+func TestParseRegion_InvalidARN_TooFewSegments(t *testing.T) {
```
Comment (musa-asad) [inline translator/util/ecsutil/ecsutil_test.go]: nit - worth adding the `< 4` boundary edges: exactly 3 segments (`arn:aws:ecs`) should hit the guard and exactly 4 (`arn:aws:ecs:us-east-1`) should pass it.

File: translator/util/ecsutil/ecsutil_test.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseRegion_ValidARN_ShortFormat(t *testing.T) {
+	e := &ecsUtil{}
+	em := &ecsMetadataResponse{TaskARN: "arn:aws:ecs:us-east-1:123456789012:task/abc123"}
```
Comment (musa-asad) [inline translator/util/ecsutil/ecsutil_test.go]: nit - `arn:aws:ecs::123456789012:task/abc` passes the guard but still sets `Region` to empty — worth a case asserting that behavior?

File: translator/util/ecsutil/ecsutil_test.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+package ecsutil
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseRegion_ValidARN_ShortFormat(t *testing.T) {
```
Comment (musa-asad) [inline translator/util/ecsutil/ecsutil_test.go]: nit - these four could collapse into one table-driven test with `t.Run`, making new boundary cases one-line additions.

**Review by musa-asad — DISMISSED** [review]

**Review by musa-asad — APPROVED** [review]

**Review by okankoAMZ — APPROVED** [review]
